src/D2L.Bmx/Program.cs

```diff
@@ -116,32 +120,41 @@
 
 rootCommand.Add( printCommand );
 
-var writeCommand = new Command( "write", "Write to AWS credentials file" )
-	{
-		accountOption,
-		durationOption,
-		orgOption,
-		writeOutputOption,
-		profileOption,
-		roleOption,
-		userOption,
-	};
+var writeCommand = new Command( "write", "Write to AWS credentials file" ) {
+	accountOption,
+	durationOption,
+	orgOption,
+	writeOutputOption,
+	profileOption,
+	roleOption,
+	userOption,
+	nonInteractiveOption,
+};
 
 writeCommand.SetHandler( async ( InvocationContext context ) => {
 	var consolePrompter = new ConsolePrompter();
+	var config = new BmxConfigProvider().GetConfiguration();
```

Comment:
it's a lot easier to inject the config itself (rather than the config provider) into all objects that need it, so we can avoid reading the config files multiple times.